File: fixtures/lens-coverage-workspace/js-messaging/src/aws-sdk-client.ts
import {
    SQSClient,
    SendMessageCommand,
    ReceiveMessageCommand,
} from '@aws-sdk/client-sqs';
import {
    SNSClient,
    PublishCommand,
} from '@aws-sdk/client-sns';
import {
    S3Client,
    PutObjectCommand,
} from '@aws-sdk/client-s3';
import {
    DynamoDBClient,
    PutItemCommand,
} from '@aws-sdk/client-dynamodb';

const sqs = new SQSClient({});
const sns = new SNSClient({});
const s3 = new S3Client({});
const ddb = new DynamoDBClient({});

sqs.send(new SendMessageCommand({ QueueUrl: 'https://sqs.us-east-1.amazonaws.com/123/orders-js', MessageBody: 'hi' }));
sqs.send(new ReceiveMessageCommand({ QueueUrl: 'https://sqs.us-east-1.amazonaws.com/123/orders-js' }));
sns.send(new PublishCommand({ TopicArn: 'arn:aws:sns:us-east-1:123:orders-js-topic', Message: 'hi' }));
s3.send(new PutObjectCommand({ Bucket: 'orders-js-bucket', Key: 'k.json' }));
ddb.send(new PutItemCommand({ TableName: 'OrdersJsTable', Item: {} }));

File: fixtures/lens-coverage-workspace/java-messaging/src/main/java/example/AwsSdkClient.java
package example;

import com.amazonaws.services.sqs.AmazonSQS;
import com.amazonaws.services.sns.AmazonSNS;
import software.amazon.awssdk.services.sqs.SqsClient;
import software.amazon.awssdk.services.sns.SnsClient;

public class AwsSdkClient {
    private SqsClient sqsV2;
    private AmazonSQS sqsV1;
    private SnsClient snsV2;
    private AmazonSNS snsV1;

    public void useSqs() {
        sqsV2.sendMessage("https://sqs.us-east-1.amazonaws.com/123/orders");
        sqsV1.sendMessage("https://sqs.us-east-1.amazonaws.com/123/orders-v1");
    }

    public void useSns() {
        snsV2.publish("arn:aws:sns:us-east-1:123:orders-topic");
        snsV1.publish("arn:aws:sns:us-east-1:123:orders-topic-v1");
    }

    public void s3Ops() {
        // legacy positional shape — exercises aws-sdk-java-s3-put / aws-sdk-java-s3-get token-line
        s3.putObject("my-bucket", "orders/2024.json");
        s3.getObject("my-bucket", "orders/2024.json");
    }

    public void dynamoOps() {
        // exercises aws-sdk-java-dynamodb-call token-line
        ddb.putItem(b -> b.tableName("OrdersTable"));
        ddb.getItem(b -> b.tableName("OrdersTable"));
    }

    private Object s3;
    private Object ddb;
}
